spec/tbd_tests_spec.rb

```diff
@@ -2251,4 +2251,218 @@
+    identifier = "TEMPORARY attic setpoint schedule"
+    sched = OpenStudio::Model::ScheduleCompact.new(model)
+    sched.setName(identifier)
+    expect(sched.constantValue.empty?).to be(true)
+    expect(tstat.setHeatingSetpointTemperatureSchedule(sched)).to be(true)
+
+    # After the addition.
+    expect(tstat.getHeatingSchedule.empty?).to be(false)
+    expect(tstat.getCoolingSchedule.empty?).to be(true)
+    heat = TBD.maxHeatScheduledSetpoint(zone)
+    expect(heat.nil?).to be(false)
+    expect(heat.is_a?(Hash)).to be(true)
+    expect(heat.key?(:spt)).to be(true)
+    expect(heat.key?(:dual)).to be(true)
+    expect(heat[:spt].nil?).to be(true)
+    expect(heat[:dual]).to be(true)
+    expect(TBD.plenum?(attic, loops, setpoints)).to be(true) # works ...
+
+    argh[:option] = "code (Quebec)"
+    json = TBD.process(model, argh)
+    expect(json.is_a?(Hash)).to be(true)
+    expect(json.key?(:io)).to be(true)
+    expect(json.key?(:surfaces)).to be(true)
+    io       = json[:io      ]
+    surfaces = json[:surfaces]
+    expect(TBD.error?).to be(true)
+    expect(TBD.logs.empty?).to be(false)
+
+    # The incomplete (temporary) schedule triggers a non-FATAL TBD error.
+    TBD.logs.each do |log|
+      expect(log[:message].include?("Empty '"))
+      expect(log[:message].include?("values' (OSut::scheduleCompactMinMax)"))
+    end
+
+    expect(io.nil?).to be(false)
+    expect(io.is_a?(Hash)).to be(true)
+    expect(io.empty?).to be(false)
+    expect(io.key?(:edges)).to be(true)
+    expect(surfaces.nil?).to be(false)
+    expect(surfaces.is_a?(Hash)).to be(true)
+    expect(surfaces.size).to eq(18)
+
+    surfaces.values.each do |surface|
+      expect(surface.is_a?(Hash)).to be(true)
+      expect(surface.key?(:conditioned)).to be(true)
+      expect(surface.key?(:deratable)).to be(true)
+      expect(surface.key?(:construction)).to be(true)
+      expect(surface.key?(:ground)).to be(true)
+      next     if surface[:ground]
+      next unless surface[:deratable]
+
+      id = surface[:construction].nameString
+      ok = id.include?("BTAP-Ext-Roof") || id.include?("BTAP-Ext-Wall")
+      expect(ok).to be(true)
+    end
+
+    # Once done, ensure temporary schedule is dissociated from the thermostat
+    # and deleted from the model.
+    tstat.resetHeatingSetpointTemperatureSchedule
+    expect(tstat.getHeatingSchedule.empty?).to be(true)
+    sched2 = model.getScheduleByName(identifier)
+    expect(sched2.empty?).to be(false)
+    sched.remove
+    sched2 = model.getScheduleByName(identifier)
+    expect(sched2.empty?).to be(true)
+    heat = TBD.maxHeatScheduledSetpoint(zone)
+    expect(heat.nil?).to be(false)
+    expect(heat.is_a?(Hash)).to be(true)
+    expect(heat.key?(:spt)).to be(true)
+    expect(heat.key?(:dual)).to be(true)
+    expect(heat[:spt].nil?).to be(true)
+    expect(heat[:dual]).to be(false)
+    expect(TBD.plenum?(attic, loops, setpoints)).to be(false) # as before ...
```

Comment:
Once TBD is done derating surfaces, users can always _reset_ the attic thermostat's heating schedule, then _removing_ it altogether from the model. Beware: the attic space would subsequently be identified by TBD as an _unconditioned_ attic again. Caution remains warranted, but the tests nonetheless illustrate how to alter TBD's interpretation of un/conditioned spaces (usually unoccupied), and at different stages of model completion.